src/Common/Hydrator/ObjectMap.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * SocialConnect project
+ * @author: Patsura Dmitry @ovr <talk@dmtry.me>
+ */
+
+namespace SocialConnect\Common\Hydrator;
+
+class ObjectMap
```

Comment:
Yeap, you are right. I am on the way to rethink hydration
Thinking about start using `array` for response hydration and write hydrator for `array` -> `User`

Because arrays are more optimized as performance and lower memory usage then object (as it was in PHP 5.4 :D)

Reply:
Concrete class instances are actually more memory efficient than arrays, though not `stdClass` instances :)

So JSON encoding response to array and then hydrating the `User` instance sounds good.

Reply:
Given that the hydration process is simple mapping of response values to `User`'s properties, you could do away with separate hydrator altogether and handle the mapping through `User`'s constructor itself. 

Something like  `$user = new User($responseArray, $mapArray);`.

Reply:
So, I migrated to `array` but I am continue to use hydrator instead of putting the black hole inside `User`'s `__constructor` 😄